### PR TITLE
Add Mullvad's DoH AnyCast DNS

### PR DESCRIPTION
**Pull request by Dyras**
Hello! [I'm back!](https://github.com/celzero/rethink-app/pull/777)

This commit adds 3 of Mullvad's DoH DNS.

Mullvad's regular DNS
Mullvad's AdBlock DNS
Mullvad's Base DNS (Which I named Mullvad DNS Anti-Malware/AdBlock due to the stupid name)

They have two more services, but I don't think they are as high in demand so to speak.

Info about the DNS:
https://mullvad.net/sv/help/dns-over-https-and-dns-over-tls/

### CODE REVIEW

**Comment by shuvashish76** [comment]
I think we should use 2 variants. 3 unnecessary as we already have option for custom field.
1. PlayStore - only `vanilla` (see #996)
2. F-Droid - `vanilla` and either `ad-block` or `base`

Note that our intention of adding Mullvad to default list should be giving users privacy friendly "audited" DNS than giving them _more DNS_ choices.

File: app/src/main/java/com/celzero/bravedns/database/AppDatabase.kt [hunk 1]
```diff
@@ -378,6 +378,16 @@ abstract class AppDatabase : RoomDatabase() {
                         execSQL(
                             "INSERT OR REPLACE INTO DoHEndpoint(dohName,dohURL,dohExplanation, isSelected,isCustom,modifiedDataTime,latency) values('Quad9 Secure','https://dns.quad9.net/dns-query','Quad9 routes your DNS queries through a secure network of servers around the globe.',0,0,0,0)"
                         )
+                        execSQL(
+                            "INSERT OR REPLACE INTO DoHEndpoint(dohName,dohURL,dohExplanation, isSelected,isCustom,modifiedDataTime,latency) values('Mullvad DNS','https://dns.mullvad.net/dns-query','Mullvad's DNS server is a no-logging service, usable even if you're not a Mullvad customer.',0,0,0,0)"
```
Comment (shuvashish76) [inline app/src/main/java/com/celzero/bravedns/database/AppDatabase.kt:382]: > usable even if you're not a Mullvad customer.

No need to mention that to RDNS users. Also using "_Mullvad's DNS_" keyword is redundant as it's already mentioned in the title.
I think better would be something like "Audited DNS server with no-logging of user activity policy."

**Comment by ignoramous** [comment]
Thanks for this.

>  Mullvad's AdBlock DNS

As pointed out, please leave the adblock endpoint out (for simplicity's sake, could leave it out for all versions). Having the adblock endpoint is a violation of play store policies.

If you're wondering about Rethink's own content-blocking resolvers, we don't really advertise "adblocking" but rather "privacy" / "security" / "parental control", and got it past the play store review only a year or so ago.

And so: let's add just the anti-malware Mullvad endpoint?

**Comment by Dyras** [comment]
Hello all! Thanks for the comments.

@shuvashish76 I can change that text, no problem!
@ignoramous Ah, I see. I'll remove the default one and the adblocking one when houssainmod responds :)

**Review by hussainmohd-a — CHANGES_REQUESTED** [review]
Thank you for the pull request. However, there are some changes that need to be made in order to add the new endpoints properly. Currently, the `addMoreDohToList()` function is part of an older migration. To incorporate these changes as part of version v055b, we need to create a new migration. Let's call it something like **MIGRATION_20_21** and place the insert script there.

The current migration version for v055a is 20, so the new changes should be part of migration 21 to maintain a logical sequence. You can find the relevant code [here](https://github.com/celzero/rethink-app/blob/main/app/src/main/java/com/celzero/bravedns/database/AppDatabase.kt#L93).

Let me know for anything

**Comment by ignoramous** [comment]
If the migration bits are a bit too much, just wait this out a bit, and we'll add Mullvad's DoT endpoints (not DoH) in the app (by default). DoT is coming in `v055b`.

**Comment by Dyras** [comment]
I looked at it a bit and it feels awfully convoluted at the moment. I'll happily wait, no rush :)

**Comment by ignoramous** [comment]
We've added Mullvad as DoT (not DoH) in `v055c` as there are enough preset DoH servers already. Closing this PR... feel free to reopen.

https://github.com/celzero/rethink-app/blob/4ca6065bfbf6475e69330982cda93d7207b56c86/app/src/main/java/com/celzero/bravedns/database/AppDatabase.kt#L839-L844

**Pull request closed without merging**